plugin/src/main/java/io/jenkins/plugins/analysis/core/model/ResultAction.java

```diff
@@ -350,6 +353,25 @@ public String resetReference() {
         return "{}";
     }
 
+    /**
+     * Displays a badge if there are issues.
+     *
+     * <p>
+     * Only for use in Jelly.
+     *
+     * @return the badge or {@code null} if there are no issues
+     */
+    @Restricted(DoNotUse.class)
+    public Badge getBadge() {
+        var warningActionsCount = getResult().getTotalSize();
+
+        if (warningActionsCount == 0) {
+            return null;
+        }
+
+        return new Badge(String.valueOf(warningActionsCount), Messages.ResultAction_Badge(warningActionsCount), Badge.Severity.WARNING);
```

Comment:
Not a big thing but a `Badge` in core should have a constructor with a numeric value that does the `String.value.of` internally.